scvi/model/base/_pyromixin.py

```diff
@@ -0,0 +1,593 @@
+        """
+        Get many samples from posterior distribution.
+
+        Parameters
+        ----------
+        args
+            arguments to model and guide
+        kwargs
+            arguments to model and guide
+        show_progress
+            show progress bar
+
+        Returns
+        -------
+        Dictionary with array of samples for each variable
+        dictionary {variable_name: [array with samples in 0 dimension]}
+
+        """
+
+        samples = self._get_one_posterior_sample(
+            args, kwargs, return_sites=return_sites, sample_observed=return_observed
+        )
+        samples = {k: [v] for k, v in samples.items()}
+
+        for _ in tqdm(
+            range(1, num_samples),
+            disable=not show_progress,
+            desc="Sampling global variables, sample: ",
+        ):
+            # generate new sample
+            samples_ = self._get_one_posterior_sample(
+                args, kwargs, return_sites=return_sites, sample_observed=return_observed
+            )
+
+            # add new sample
+            samples = {k: samples[k] + [samples_[k]] for k in samples.keys()}
+
+        return {k: np.array(v) for k, v in samples.items()}
+
+    def _posterior_samples_full_data(self, use_gpu: bool = True, **sample_kwargs):
+        """
+        Generate samples from posterior distribution using all data
+
+        Parameters
+        ----------
+        sample_kwargs
+            arguments to _get_posterior_samples
+
+        Returns
+        -------
+        dictionary {variable_name: [array with samples in 0 dimension]}
+
+        """
+
+        self.module.eval()
+        gpus, device = parse_use_gpu_arg(use_gpu)
+
+        args, kwargs = self.module.model._get_fn_args_full_data(self.adata)
+        args = [a.to(device) for a in args]
+        kwargs = {k: v.to(device) for k, v in kwargs.items()}
+        self.to_device(device)
+
+        samples = self._get_posterior_samples(args, kwargs, **sample_kwargs)
+
+        return samples
+
+    def _posterior_samples_minibatch(self, use_gpu: bool = True, **sample_kwargs):
+        """
+        Generate samples of the posterior distribution of each parameter, separating local (minibatch) variables
+        and global variables, which is necessary when performing minibatch inference.
+
+        Note for developers: requires model class method which lists observation/minibatch plate
+        variables (self.module.model.list_obs_plate_vars()).
+
+        Parameters
+        ----------
+        use_gpu
+            Bool, use gpu?
+
+        Returns
+        -------
+        dictionary {variable_name: [array with samples in 0 dimension]}
+
+        """
+
+        gpus, device = parse_use_gpu_arg(use_gpu)
+
+        self.module.eval()
+
+        train_dl = AnnDataLoader(self.adata, shuffle=False, batch_size=self.batch_size)
+        # sample local parameters
+        i = 0
+        with tqdm(train_dl, desc="Sampling local variables, batch: ") as tqdm_dl:
```

Comment:
we have a custom wrapper for tqdm. See here:

https://github.com/YosefLab/scvi-tools/blob/6dae6482efa2d235182bf4ad10dbd9483b7d57cd/scvi/data/_built_in_data/_download.py#L43-L45